website/blog/2023-04-01-databend-weekly-87.md

```diff
@@ -0,0 +1,181 @@
+
+    ...
+
+    for mut tasks in [tasks, polling_tasks] {
+        tasks.sort_by(|l, r| Ord::cmp(&l.stack_frames.len(), &r.stack_frames.len()));
+
+        for item in tasks.into_iter().rev() {
+            for frame in item.stack_frames {
+                writeln!(output, "{}", frame).unwrap();
+            }
+
+            writeln!(output).unwrap();
+        }
+    }
+```
+
+Calling the `async_backtrace::taskdump_tree` function can obtain information about the asynchronous task tree (collected by `#[async_backtrace::framed]`). Tasks are divided into regular tasks and polling tasks (marked with `[POLLING]`). Record the stack information for each task, and output it to a string sorted by stack depth.
+
+To learn more about how it works, refer to the following link:
+
+- [PR | feat(query): try support dump running async task stack](https://github.com/datafuselabs/databend/pull/10830)
+
+### Meet Databend in Jupyter Notebook
+
+As described in [Doc | Visualization Databend Data in Jupyter Notebook](https://databend.rs/doc/integrations/gui-tool/jupyter), we can use Jupyter Notebook to explore data in Databend.
+
+![](https://datafuse-1253727613.cos.ap-hongkong.myqcloud.com/integration/integration-jupyter-databend.png)
+
+However, through the trick of magic, [ipython-sql](https://github.com/catherinedevlin/ipython-sql) provides another interesting way to make SQL queries look like they are running in SQL cells and maintain seamless integration with Python.
+
+**Install Dependencies**
+
+```bash
+pip install ipython-sql databend-sqlalchemy
+```
+
+**Work with Jupyter Notebook**
+
+```sql
+In [1]: %load_ext sql
+
+In [2]: %%sql databend://{username}:{password}@{host_port_name}/{database_name}?secure=false
+   ...: SHOW DATABASES;
+
+In [3]: result = %%sql SELECT * FROM numbers(100);
+
+In [4]: %matplotlib inline
+   ...: df = result.DataFrame()
+   ...: df.plot()
+```
+
+## Highlights
+
+Here are some noteworthy items recorded here, perhaps you can find something that interests you.
+
+- *[Connecting to Databend with DBeaver](https://databend.rs/blog/dbeaver)*, either by selecting the pre-configured MySQL driver or adding the Databend JDBC driver.
+- Databend provides integration with Redash to help you gain better insights into your data. *[Doc | Integrations - Redash](https://databend.rs/doc/integrations/gui-tool/redash)*
+- Master how to display information of columns in a given table. *[Doc | SHOW COLUMNS](https://databend.rs/doc/sql-commands/show/show-full-columns)*
+- Databend now supports `generate_series` and `range` table functions.
+- Databend now supports the `ai_embedding_vector` function, which returns a 1536-dimensional f32 vectors generated by the OpenAI Embeddings API.
+- Databend added support for `[CREATE | DROP | SHOW] SHARE ENDPOINT` DDL.
+
+## What's Up Next
+
+We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.
+
+### Collect Sled Metrics
```

Comment:
I dont understand here, we already use sled to capture metrics?

Reply:
Not really, we just need to collect metrics from the sled.

Reply:
**Collect Metrics from Sled**